lib/cadet/assessments/assessments.ex

```diff
@@ -351,8 +352,13 @@ defmodule Cadet.Assessments do
            {:is_open?, true} <- is_open?(question.assessment),
            {:ok, submission} <- find_or_create_submission(user, question.assessment),
            {:status, true} <- {:status, submission.status != :submitted},
-           {:ok, _} <- insert_or_update_answer(submission, question, raw_answer) do
+           {:ok, answer} <- insert_or_update_answer(submission, question, raw_answer) do
         update_submission_status(submission, question.assessment)
+
+        if answer.comment == "" or answer.comment == nil do
+          Room.create_rooms(submission, answer, user)
+        end
```

Comment:
Actually, can this be placed where a submission occurs?

Reply:
May I clarify what you mean? Do you mean after submissions are finalised?

Reply:
Yes. I think it is fine to leave the function here for now. But ultimately room creation should be done when a submission is submitted right?

Reply:
Sure. I'll place it [here](https://github.com/source-academy/cadet/blob/master/lib/cadet/jobs/autograder/grading_job.ex#L63) then? Since this is where all submissions flow through, whether it's submitted early or submitted by the autograder. Quite ugly tho :/